homeassistant/components/hlk_sw16/__init__.py

```diff
@@ -2,134 +2,133 @@
 import logging
 
 from hlk_sw16 import create_hlk_sw16_connection
-import voluptuous as vol
 
 from homeassistant.const import (
     CONF_HOST,
     CONF_NAME,
     CONF_PORT,
-    CONF_SWITCHES,
     EVENT_HOMEASSISTANT_STOP,
 )
 from homeassistant.core import callback
-import homeassistant.helpers.config_validation as cv
-from homeassistant.helpers.discovery import async_load_platform
 from homeassistant.helpers.dispatcher import (
     async_dispatcher_connect,
     async_dispatcher_send,
 )
 from homeassistant.helpers.entity import Entity
 
+from .const import (
+    CONNECTION_TIMEOUT,
+    DEFAULT_KEEP_ALIVE_INTERVAL,
+    DEFAULT_RECONNECT_INTERVAL,
+)
+
 _LOGGER = logging.getLogger(__name__)
 
 DATA_DEVICE_REGISTER = "hlk_sw16_device_register"
-DEFAULT_RECONNECT_INTERVAL = 10
-DEFAULT_KEEP_ALIVE_INTERVAL = 3
-CONNECTION_TIMEOUT = 10
-DEFAULT_PORT = 8080
+DATA_DEVICE_LISTENER = "hlk_sw16_device_listener"
 
-DOMAIN = "hlk_sw16"
 
-SWITCH_SCHEMA = vol.Schema({vol.Optional(CONF_NAME): cv.string})
-
-RELAY_ID = vol.All(
-    vol.Any(0, 1, 2, 3, 4, 5, 6, 7, 8, 9, "a", "b", "c", "d", "e", "f"), vol.Coerce(str)
-)
-
-CONFIG_SCHEMA = vol.Schema(
```

Comment:
This instantly makes the YAML configuration invalid and can potentially break existing setups.
Can we import existing configuration?

Reply:
I looked into that but it seemed quite difficult to migrate.

Reply:
Is it fine to leave migrations unsupported? I'd really prefer not to support migrations for this device due to it being difficult to map the old config format to the new one. Due to the type of device I don't think it will be that big an issue for users to manually migrate.

Reply:
I added a code suggestion to get you part of the way there.